### ISSUE TITLE
Fundamental Weapons Categorization/Changes

### ISSUE THREAD

**Issue by ZachZent** [tsunamayo/Starship-EVO#3797]
TL;DR but please do, Suggestion to A. make every weapon (future) separate instead of adding weapon changing barrels and B. organize the modules and barrels into similar purposes (ex. barrels mods the projectile - range, modules mods the weapon itself - cooling)
______________

Weapons need a lot of fixing, the model, the balance, and most importantly adding more of them. As of now however what can be fixed, or rather future proofed, is the purpose and categorization of core, barrels, modules and the distinctions/functions between the three.

Right now, the barrels and modules are an odd mixed bag of purposes. Cutting to the chase, the gatling barrel. I don’t expect you plan to add gatling capabilities for all weapons (though gatling flak does have its charm) and if you want to add similar core weapon modification barrels, it will have the same issue of working for a few but not all realistically. The more weapons added, the greater the issue, the greater the confusion. Both you and modders will find down the line that new weapons will either be inconsistent one-use barrels/modules or new individual weapons which can’t use some or all barrels/modules. For example, a rail gun (a la Expanse) style type of kinetic. Should it be a one-use barrel off the base kinetic weapon unusable by any other weapon or a new weapon that would be able to use some but not all of the options. Finally, by mixing the capabilities of barrels and modules, it reduces the total customization options. Instead, I recommend two things.

First is rather than having barrels change the function of the weapon, just have all the weapons as separate entities. Your laser and gatling laser as separate weapons. This makes it very easy for a newbie to understand and allows common customization between all weapons while allowing you to decide which specific weapons can use barrels, modules, or neither. Plus, it makes it far easier to balance the core weapons rather than jumping between weapon and barrels needing to balancing multiple mixed systems at once. Each weapon being separate would cause some codex clutter, but unlike clutter from say blocks, I don’t think anyone will mind being able to see all weapons at once and compare their uses and base stats. After all, everyone likes guns and seeing all the options with all the modifications that can be done is very appealing for both current and potential players. Then when more weapons are added in the future by you and/or by modders, they can just be added to the list for all to easily see and use.

Second for the sake of clarity, better descriptors, and overall balance, an overall decision on what impact the weapon barrels and modules will have should be decided. As said, current modules and barrels are a mixed bag of purposes which allows odd combinations, especially in the future when/if more is added. Knowing that modules affect X while barrels affect Y makes it easier to understand and to add new ones in the future. Perhaps the barrels impact the stats of the projectile such as range or effectiveness against armor while the modules impact the stats of the weapon itself such as cooling or rate of fire.

**Comment by Garrett-C** [comment]
I disagree with the suggestion to remove barrels and replace it with just a variety of different weapons. For 3 main reasons.

1. Codex space. Assuming that barrels are usable on say the kinetic weapons you end up with a smaller number of items in the codex than you would if you had a gatling laser, gatling kinetic, long range laser, long range kinetic etc. 

2. Balance. Currently the actual laser has a max length regardless of what function you assign for it. Barrels are the same but have a slight variety between types. So if each gun is it's own type that brings up a lot of balancing issues. Say the max length of the gatling gun is 6m but the long range one is 8m, which one has more damage, is it the same because they are max length? What about if you only make yout long range one 7m instead of the max 8? 

3. Control. Currently the player can choose how long to make the laser and the barrel so for a 7m long gun you could go with 4 lasers and 3 barrels, the oposite, 2 lasers and 5 barrels etc. You can fine tune how you want the weapon to work. If you oversimplify it down to a single block you loose that control.

**Comment by ZachZent** [comment]
> As an edit I just realised you actually seem to be suggesting keeping barrels but taking their functions and making weapon types for them. 

What I was suggesting is first remove the gatling and any future barrel that fundamentally changes the function of the weapon and keep it as its own separate weapon. When more weapons are added, the fewer total weapons will be able to use the gatling barrel making it a one-off use only thus confusing new players. Lets look at the confirmed weapons (ignoring missiles).
`Laser - Beam - Kinetics - Flame Thrower - Flak*`
The gatling barrel would only work on two of them, the laser and the kinetics and not the beam or flamer. When more weapons are added to the list both by Tsuna and modders, the confusion will only increase on why this barrel works on some and not others.

*Even flak which in the current model would probably be a barrel for kinetics wouldn't really be logical on any other weapon which means you have a single barrel to be used on one weapon type and none others. I can already see the question posts "why can't I put a flak barrel on my laser gun?" though laser flak does sound fun.

Another example in the post for a new weapon was an Expanse style railgun (charge up and fire single shot). This is obviously a subset of kinetics which for now I will assume will be leaning more to the higher RPM. You could have a railgun barrel, but it would only work on kinetics making it a one-weapon attachment leading to confusion and clutter.
[https://i.imgur.com/5BEA03Z.mp4](url)

> Sounds like that would just add a lot more codex clutter

Sort of yes, but slightly more clutter is better then confusing clutter. For example remove the gatling barrel and make a separate gatling laser and gatling kinetic and you get only one extra item with the benefit of clarity. I don't think a gatling flame thrower will be a thing. Maybe with the gatling weapons, they can't use any barrel like the current beam weapon for balance sake which makes far more sense then the gatling barrel only working on two weapons and none other. Or maybe they can, at least it gives Tsuna the option for customization ability.

For the flak and rail gun example, it is replacing a one-use barrel for a new weapon, equal number of items. Slightly more clutter is better then confusion and completely ignores that the "clutter" will be a list of all the guns you can use. And as clutter goes, that is one I am perfectly happy with as it is nice for new and experienced players to see all the options and is good for marketing :D

> I can see people trying to stick long range barrels on their gating laser. Or being confused as to why the gattling barrel on their normal laser is not making ir gattling.

The gatling barrel would be removed as would all future fundamentally changing barrels/modules so that point is null. And just to be clear, I'm not saying take all barrels and make a separate weapon like a long range beam and a cooling beam, that would be silly. The barrels would mostly stay as is** with the current exception of the gatling barrel and future fundamental weapon barrels like the theoretical kinetic-only flak barrel.

**To remind of point two of the suggestion, barrels and modules would be categorized into how they more specifically alter the weapon rather then mixing them around. In the example, barrels would be for modding the projectile (range) while modules would be for modding the core weapon (cooling).

**Comment by Garrett-C** [comment]
So in summary you want to make the gatling barrel into its own special weapon for each weapon type that uses that type?

How is that really any less confusing than the current system? Currently you have a weapon and a set of barrels which change how the weapons function. Your suggestion would lead to you having a standard weapon, a set of barrels whi h change how it functions and special 1 off weapons which aren't compatible with the barrel system. I would argue that's more confusing than the current solution. 

I can see the posts of people asking "Why can't I put a recoil barrel on my gatling gun?"

Surely it would be more reasonable and easier to understand to just put a "compatible weapon types" list in the description of the part in the codex? 

The issue is your suggestion creates more clutter and doesn't clear up more confusion, if anything I would argue it creates more confusion as you have more weapons that barrels can't work with.

**Comment by TIKIRobo** [comment]
I agree with Zach here, even though it might add on 5 whole new items to the codex (and replace others) it would be a lot easier for new players and for code stuff.

Instead of having lasers and kinetics that get barrels and beams that don't (which is confusing) you would just have a rail gun and a machine gun and yada yada, maybe there could be addons that would be universal but idk if thats the sort of thing Tsuna wants

**Comment by Garrett-C** [comment]
> I agree with Zach here, even though it might add on 5 whole new items to the codex (and replace others) it would be a lot easier for new players and for code stuff.
> 
> Instead of having lasers and kinetics that get barrels and beams that don't (which is confusing) you would just have a rail gun and a machine gun and yada yada, maybe there could be addons that would be universal but idk if thats the sort of thing Tsuna wants

Here's my issue though. You still have a lot of the barrels but now have more specialised weapons that can't use them. So while attempting to simplify matters you in fact make it more confusing. 

You simply end up with more weapons that can't use the barrels. I fail to see how a gatling barrel that works on 2 weapon types is more confusing than like 5 or 6 weapon types that just can't use any barrels.

**Comment by ZachZent** [comment]
> How is that really any less confusing than the current system? Currently you have a weapon and a set of barrels which change how the weapons function.

Right now it works because we have a laser and a beam. But when you start adding more weapons like a flame thrower, flak, etc, the problem will compound exactly how you claim my idea will. Weapons which can't use every barrel that can be used, more items for specialized barrels, and so on.

> Your suggestion would lead to you having a standard weapon, a set of barrels which change how it functions and special 1 off weapons which aren't compatible with the barrel system.
I fail to see how a gatling barrel that works on 2 weapon types is more confusing than like 5 or 6 weapon types that just can't use any barrels.

On the first part let me rephrase, Tsuna can make the **choice** if the "specialized weapons" can have barrels or not. In fact, that is what he is doing now with the beam which can only have modules. There would be nothing preventing Tsuna from deciding that a gatling laser can or can't use the standard barrels. If you think about it, a gatling barrel technically is a specalized weapon as is since you can't stack another barrel on top of the gatling. The main difference is that with my idea it is **all or nothing** for barrels or/and modules for all weapons rather then **some or nothing** with even more asterisks on what can be used the more weapons are added.

> Surely it would be more reasonable and easier to understand to just put a "compatible weapon types" list in the description of the part in the codex?

Laser: Restricted list
Flak barrel, railgun barrel, plasma barrel, supermegabeam barrel. It only gets worse the more weapons/barrels are added, though side-bar descriptions are always helpful regardless.

> idk if thats the sort of thing Tsuna wants

It really comes down to if Tsuna wants to keep the core-barrel-module setup or not. If he does and wants to add more weapons in the future besides the basic ones, it is better to nip this in the bud now. Especially for point 2 of the original suggestion.

**Comment by Garrett-C** [comment]
> Right now it works because we have a laser and a beam. But when you start adding more weapons like a flame thrower, flak, etc, the problem will compound exactly how you claim my idea will. Weapons which can't use every barrel that can be used, more items for specialized barrels, and so on.

And this is exactly where your suggestion baffles me. Your issue is that when more weapon types are added that can't use these barrels the system will become more confusing. That's a fair issue and one that can be looked into. Your suggestion to solve this issue though is adding more weapons that can't use these barrels... Where is the logic here? I don't disagree that the system can be improved but I don't think making the very problem you are pointing out worse is the way to achieve that.

> On the first part let me rephrase, Tsuna can make the choice if the "specialized weapons" can have barrels or not. In fact, that is what he is doing now with the beam which can only have modules. There would be nothing preventing Tsuna from deciding that a gatling laser can or can't use the standard barrels. If you think about it, a gatling barrel technically is a specalized weapon as is since you can't stack another barrel on top of the gatling. The main difference is that with my idea it is all or nothing for barrels or/and modules for all weapons rather then some or nothing with even more asterisks on what can be used the more weapons are added.

By the logic, you voice here every barrel is a specialist barrel. I haven't built many guns recently but I don't recall being able to stack recoil barrels on top of range barrels. That's part of the point though. Your solution isn't all or nothing. You said yourself that you aren't suggesting that all barrels become their own weapons. So you are just adding in a new weapon type that can't use all the barrels, which is the very issue you are outlining to begin with.

I would suggest something like this:

![Barrel image](https://user-images.githubusercontent.com/43644633/120210118-5590d400-c227-11eb-8edb-38682d5f0e8a.png)

Let's break things down a bit though.

Laser: 
Barrels - Range, Cooling, Recoil, Gatling
Notes - As is currently

Kinetic:
Barrels - Range, Cooling, Recoil, Gatling, Flak, railgun (Maybe just called rails or something)
Notes - I don't quite get why flak would be a separate gun type? It fires projectiles yes? It isn't going to need long-range or recoil addons so surely it works as a barrel for kinetic.

Beams:
Barrels - None
Notes - Same as current, could add some form of focusing barrel or something down the line, or a redirection barrel if the beam merging ever returns. 

Missiles and flamethrowers I see as being a completely different set of systems. 

I fail to see how creating a Gatling laser, Gatling kinetic and a flak kinetic that all can't use the standard barrels simplifies the system? What you do is double the number of weapon bases and now make it so that 2/6 of the weapon types can use barrels. That just makes matters more confusing if you ask me.

**Comment by TIKIRobo** [comment]
> Here's my issue though. You still have a lot of the barrels but now have more specialised weapons that can't use them. So while attempting to simplify matters you in fact make it more confusing. 

The point would be not have have some sort of half and half system, so you wouldn't have gatling blocks and barrels. most barrels would be removed and instead a laser block could be made into a "gatling laser" by slapping some fire speed addons or a flak cannon block or whatever

**Comment by Garrett-C** [comment]
> > Here's my issue though. You still have a lot of the barrels but now have more specialised weapons that can't use them. So while attempting to simplify matters you in fact make it more confusing.
> 
> The point would be not have have some sort of half and half system, so you wouldn't have gatling blocks and barrels. most barrels would be removed and instead a laser block could be made into a "gatling laser" by slapping some fire speed addons or a flak cannon block or whatever

The thing is that's not what's being suggested here. The suggestion is making gatling, flak and rail guns into thier own weapons. Zach explicitly says " I'm not saying take all barrels and make a separate weapon like a long range beam and a cooling beam, that would be silly. The barrels would mostly stay as is** with the current exception of the gatling barrel and future fundamental weapon barrels" So you end up with more 1 off weapons that can't use all the standard barrels, thus making the system more confusing.

**Comment by ZachZent** [comment]
I believe we are getting closer to the same page here

To start, let me redefine something.
Specialist barrel - Barrel that changes the fundamental function of the weapon like gatling or flak
Universal barrel - Barrel that can reasonably be used on all weapons as all it changes is one, maybe two stats like range or cooling

>  Your suggestion to solve this issue though is adding more weapons that can't use these barrels... 
So you end up with more 1 off weapons that can't use all the standard barrels, thus making the system more confusing.

The solution is first to remove all specialist barrels like gatling now and in the future. All that will remain is universal barrels such as range which can be placed on any weapon without general conflict. IE gatling with range barrel works, flak with gatling barrel doesn't. After that, it is up to Tsuna if he wants to give the weapon itself the option of having barrels, modules, both, or neither. Just like current beam weapon. Maybe the future flak gun can use the universal barrels, maybe it can't. But if Tsuna decides they can, all universal barrels will be able to be added without conflict where as if specialist barrels are still in the game, they shouldn't be able to be placed. Gatling flak as cool as it sounds isn't going to happen.

> Let's break things down a bit though. etc etc.

Exactly the problem. You have this massive list of rules and exceptions that will only get longer the more weapons and barrels are added when it is far more simple to have more individual weapons with a simple yay or nay on if it can use the universal modules or barrels. A common element for all where the current mixed system will only get more complicated on rules and exceptions the more that is added.

![image](https://user-images.githubusercontent.com/36980446/120220511-ad144d00-c1f1-11eb-8cff-7d509b1a1436.png)
This is what I am looking at. A large set of individual weapons, even though some are similar in function (basic kinetics vs flak for example) with a relatively smaller set of universal barrels and modules that can be used on some/most/all weapons determined on what Tsuna wants.

What I believe you see is the opposite with a limited number of core weapons with a larger selection of barrels/modules which to me is far more confusing if it needs a long text description on what barrels it can and can't use. Plus as more specialist barrels are added, only the core weapon will be able to use the universal barrels. The further it goes, the more confusing it gets.

**Comment by Garrett-C** [comment]
> I believe we are getting closer to the same page here
> 
> To start, let me define something.
> Specialist barrel - Barrel that changes the fundamental function of the weapon like gatling to laser/kinetic or flak to kinetic
> Universal barrel - Barrel that can reasonably be used on all weapons as all it changes is one, maybe two stats like range or cooling
> 
> > Your suggestion to solve this issue though is adding more weapons that can't use these barrels...
> > So you end up with more 1 off weapons that can't use all the standard barrels, thus making the system more confusing.
> 
> The solution is first to remove all specialist barrels like gatling now and in the future. All that will remain is universal barrels such as range which can be placed on any weapon without general conflict. IE gatling with range barrel works, flak with gatling barrel doesn't. After that, it is up to Tsuna if he wants to give the weapon itself the option of having barrels, modules, both, or neither. Just like current beam weapon. Maybe the future flak gun can use the universal barrels, maybe it can't. But if Tsuna decides they can, all universal barrels will be able to be added without conflict where as if specialist barrels are still in the game, they shouldn't be able to be placed. Gatling flak as cool as it sounds isn't going to happen.

That's part of the issue though. There is no reasonable way, aesthetically or balance-wise, that you could have something like a range barrel on a Gatling gun. 1. The model just doesn't line up and short of adding Gatling specific universal barrels (Defeating the whole point of the barrel being universal) the models just wouldn't line up. 2. Balance wise you make the weapon very unbalanced. Why run a laser with a range addon when you could have the gatling with the range addon. Plus the smallest gatling gun you can make currently with a 1m block is 2m. Your system would result in it having a smaller form factor which only makes it more overpowered.

> > Let's break things down a bit though. etc etc.
> 
> Exactly the problem. You have this massive list of rules and exceptions that will only get longer the more weapons and barrels are added when it is far more simple to have more individual weapons with a simple yay or nay on if it can use the universal modules or barrels. A common element for all where the current mixed system will only get more complicated on rules and exceptions the more that is added.

I can agree with your suggestion that it's a problem but your solution only adds to the problem. In my opinion beams being the only type not to use barrels and kinetic maybe having 1 or 2 specialty barrels isn't a complex system to understand. There are a couple of rules to it but they are simple. Barrels on weapons is the norm and there is one exception.

> ![image](https://user-images.githubusercontent.com/36980446/120221263-f0bb8680-c1f2-11eb-9c1f-a890de443554.png)
> This is what I am looking at. A larger set of individual weapons, ever expandable, even though some are similar in form (basic kinetics vs flak for example) with a relatively smaller set of universal barrels and modules that can be used on some/most/all weapons determined on what Tsuna wants. Obviously more barrels can be added*
> 
> What I believe you see is the opposite with a limited number of core weapons with a larger selection of barrels/modules which to me is far more confusing if it needs a long text description on what barrels it can and can't use. Plus as more specialist barrels are added, only the core weapon will be able to use the universal barrels. The further it goes, the more confusing it gets.

I fundamentally disagree. More weapons that can rarely use the barrels is more confusing than fewer weapons that can mostly use them. 

Maybe there is a way to have super prefab weapons but you lose the freedom to customise them. I get wanting to smooth things out to make it more user-friendly and I don't disagree that there is room to do that but weapons should feel like lego not megablocks. 

I do agree with your basic concept that things can be made more clear for the player, I just don't think your solution is the way to solve it.

**Comment by ZachZent** [comment]
> 1. The model just doesn't line up and short of adding Gatling specific universal barrels (Defeating the whole point of the barrel being universal) the models just wouldn't line up. 2. Balance wise you make the weapon very unbalanced. Why run a laser with a range addon when you could have the gatling with the range addon. 

Models will be changing, I believe Tsuna said that awhile back. Especially considering the modules being so extended out from the gun so that isn't an issue. As for the balance, I don't see an issue with a longer ranged gatling given how at longer ranges it will be more inaccurate anyway at a cost of heat and/or power. But that is up to Tsuna to decide which weapons can get the barrels or not and how it will be balanced.

> In my opinion beams being the only type not to use barrels and kinetic maybe having 1 or 2 specialty barrels isn't a complex system to understand. There are a couple of rules to it but they are simple. Barrels on weapons is the norm and there is one exception.

Ok, there is a fundamental misunderstanding here. Right now it is "works fine". We have two core weapons and one barrel. That is easy to understand. But this won't always be the case, hence deciding how things **should** be organized now rather then down the line. What happens when you have 4 core weapons and 6 different barrels, half universal and half specialized? How about 6 and 10? There are loads of sci-fi weapons people will want added beyond the two/three we have now and the four/five confirmed total weapons. The more added, the longer the compatibility list gets for every weapon and every barrel.

In comparison with my idea, you would know from the start that 3 weapons can use the universal barrel set and 2 can't. **All or nothing, not some or nothing.** Your thoughts works if the weapons list is kept simple forever. If that is to be then so be it, but I and many others will be very disappointed in the lack of weapons and variety. Something that may not be a core focus of the game, but is essential to have.

Past that there isn't more to say until another person comments their thoughts, but it is up to Tsuna. Does he want more core weapons with fewer universal barrels or fewer core weapons with more barrels both universal and specialized.

**Comment by Garrett-C** [comment]
> > 1. The model just doesn't line up and short of adding Gatling specific universal barrels (Defeating the whole point of the barrel being universal) the models just wouldn't line up. 2. Balance wise you make the weapon very unbalanced. Why run a laser with a range addon when you could have the gatling with the range addon.
> 
> Models will be/can be changed, I believe Tsuna said that awhile back. Especially considering the modules being so extended out from the gun so that isn't an issue. As for the balance, I don't see an issue with a longer ranged gatling given how at longer ranges it will be more inaccurate anyway at a cost of heat and/or power. But that is up to Tsuna to decide which weapons can get the barrels or not and how it will be balanced.

They can be changed yes, but given how a Gatling weapon looks in any form of media with the multiple barrels and how just a standard 1 barrel laser looks I can't see how you could stick a 1 barrel range addon onto a 4 barrel Gatling weapon. So all weapons would need to become 1 barrel, which again removes the visual differences of them. Currently, you can tell if a weapon is a Gatling weapon or a long-range weapon etc by looking at it. Having a universal barrel removes that aspect of it.

Gattling having it's high rate of fire works well because the spread makes it useless for long-range, add a long-range barrel onto it and why use any other weapon.

Also I want this cleared up. In your system where Gatling is it's own type do you propose it has access to all other barrels or not? You seem to change your mind on this depending on which question I ask so can we set your opinion on this out clearly so I know what you are actually saying. The "Tsuna can choose what guns get what barrels" point just confuses me as to what you actually are suggesting.

If all weapons can use all barrels and you have Gatling guns that are too powerful because of barrel addons
If all weapons can't use all barrels then you have the same problem we currently have but worse.

> > In my opinion beams being the only type not to use barrels and kinetic maybe having 1 or 2 specialty barrels isn't a complex system to understand. There are a couple of rules to it but they are simple. Barrels on weapons is the norm and there is one exception.
> 
> Ok, there is a fundamental misunderstanding here. Right now it "works fine". We have two core weapons, one specalized barrel, and three universal barrels. That is easy to understand. But this won't always be the case, hence deciding how things **should** be organized now rather then down the line. What happens when you have 4 core weapons and 6 different barrels, half universal and half specialized? How about 6 and 10? There are loads of sci-fi weapons people will want added beyond the two/three we have now and the four/five confirmed total weapons. The more added, the longer the compatibility list gets for every weapon and every barrel.
> 
> In comparison with my idea, you would know from the start that 3 weapons can use the universal barrel set and 2 can't. **All or nothing, not some or nothing.** Your thoughts works if the weapons list is kept simple forever. If that is to be then so be it, but I and many others will be very disappointed in the lack of weapons and variety. Something that may not be a core focus of the game, but is essential to have.

I get what you are saying but lets look at the weapons we are anticipating ok.

Kinetics - Most if not all current barrels could be fine here.

Railguns - Clearly this could be a barrel addon for a kinetic weapon

Flak - Again can be an addon for a kinetic weapon

Flamethrower - I honestly don't see the point in this as a weapon but assuming we get it I can't see it being anything like the current weapons at all as it is a fundamentally different sort of weapon. 

Missiles - Same as the flamethrower, fundamentally different so would use a completely different system.

So you end up with Lasers, beams and kinetics as the main types and missiles and flamethrowers as completely different sub types with their own systems. Kinetics have access to all barrels, lasers have access to all but flack and rail gun, beams have none. I don't see this being difficult. 

Additionally, you're all or nothing thing is blatantly not all or nothing. It results in a large variety of weapons, most of which being unable to use barrels. It's all or nothing on a weapon by weapon basis but not when it comes to weapons as a whole. As a whole, it just creates more confusion as now you have x number of weapon types buy only like 2 that use any of the addon systems.

The fundamental issue here is your solution just seems to compound the problem.

**Comment by ZachZent** [comment]
Ok from the top.

You have many different weapons. Laser, beam, kinetics, flak, sonic weapons, flame thrower (which isn't a separate system as like every other gun it has ammo and projects from a single point), railgun, harpoon, cannon, lightning gun and any other weapon you can think of that Tsuna or modders will add. Each of them is its own separate core, core being the base weapon you place. There are more weapons then just bullets, lasers, and their derivatives especially in an arcady sci-fi game.

Barrels won't change how the gun fundamentally functions, it won't change the laser into a gatling laser. The reason for this is of the list of random weapons above, only the laser and kinetic would be able to logically use this where the flak, flamer, harpoon, and cannon would not. Or a high explosive barrel which would only be able to logically work on the flak and maybe kinetics. More specializations that are considered, the more amount of specialized barrels you would need if you only have a small set of core weapon types. 

Instead there would just be a single set of universal barrels that can logically go on any weapon. Range works with all these weapons, same with cooling. Even on beam weapon(s) it makes sense to use an increased range barrel at some balancing cost for example as duration of the beam. However due to reasons of balancing or whatever else that Tsuna decides, some weapons may not be able to use barrels such as current beams. This will be obvious in the menu that they aren't able to use the universal barrels. In comparison in the current system, a sonic weapon would be able to some of the specialized barrels but not all of them. Perhaps it shares an amplifier barrel with the flame thrower which would require each weapon to have a large list of "approved" barrels and modules. The more barrels and modules added, the longer these lists.

The simple advantage of this system is that you see all the weapon options in front of you and what each of them does. Barrels and modules are there to slightly alter the weapon to your liking, if you want more range at the cost of X or more damage at the cost of Y.

> As a whole, it just creates more confusion as now you have x number of weapon types buy only like 2 that use any of the addon systems.
And btw this remains the same in either system. Beam doesn't have barrels in current or my idea. So the issue is the same except in current you will have weapons that can't use any barrels, weapons that can only use some of the specialized barrels, and weapons that can use the universal barrels and only some of the specialized barrels. To figure out what weapons can use what barrels, you either have to read though a list or start picking ones at random and testing (the noobie approach) where in my idea you know by shot description if the weapon can use the universal barrel or not.

**Comment by Garrett-C** [comment]
> Ok from the top.
> 
> You have many different weapons. Laser, beam, kinetics, flak, sonic weapons, flame thrower (which isn't a separate system as like every other gun it has ammo and projects from a single point), railgun, harpoon, cannon, lightning gun and any other weapon you can think of that Tsuna or modders will add. Each of them is its own separate core, core being the base weapon you place. There are more weapons then just bullets, lasers, and their derivatives especially in an arcady sci-fi game.

1. Some of these weapon types, namely sonic, lightning and harpoon, are not reasonable to be incorporated into the game. 
2. Flamethrowers and missiles are completely different from standard weapons. 
3. The core systems of the game shouldn't me built around mod. Will a modder add a harpoon? Maybe, should the games weapon systems be built around the possibility that someone does that no. If you add a mod to your game you are intentionally deviating from the developer's intended experience. If that makes the weapon system more confusing then that's on you.
4. There are more weapons sure but what value do they add to the game. Kinetics, beams, lasers and missiles all add something different and valuable, lightning or sonic not so much.

> Barrels won't change how the gun fundamentally functions, it won't change the laser into a gatling laser. The reason for this is of the list of random weapons above, only the laser and kinetic would be able to logically use this where the flak, flamer, harpoon, and cannon would not. Or a high explosive barrel which would only be able to logically work on the flak and maybe kinetics. More specializations that are considered, the more amount of specialized barrels you would need if you only have a small set of core weapon types.

Barrels should be able to change the gun's fundamental function though. It adds depth to the building side of the game that is needed to ensure that shipbuilding and designing is a worthwhile experience. If the addons you add to your weapons provide no difference other than small stat buffs then they are a pointless addition to the game experience. 

Weapons like Flak, railgun and Gatling are fundamentally laser or kinetic weapons with a different purpose. So a core weapon type that can be turned into these other types allows for a much more involved building experience that still makes sense. There is no need for 100 different specialisations in the game. 

> Instead there would just be a single set of universal barrels that can logically go on any weapon. Range works with all these weapons, same with cooling. Even on beam weapon(s) it makes sense to use an increased range barrel at some balancing cost for example as duration of the beam. However due to reasons of balancing or whatever else that Tsuna decides, some weapons may not be able to use barrels such as current beams. This will be obvious in the menu that they aren't able to use the universal barrels. In comparison in the current system, a sonic weapon would be able to some of the specialized barrels but not all of them. Perhaps it shares an amplifier barrel with the flame thrower which would require each weapon to have a large list of "approved" barrels and modules. The more barrels and modules added, the longer these lists.

The universal barrel doesn't work though for several main reasons.
1. The barrel model would need to some how adjust to the model of the gun which would be impossible to show in the codex, think how gatling is 4 barrels but lasers only have 1. So a range addon barrel that would be universal to both would need to somehow be both 4 barrels and 1 barrel.
2. The universal barrel look eliminates the aesthetic customisation that the barrels provide. It also limits how much information you can get from a weapon by looking at it. If a ship is using a specialised weapon type, like a gattling gun or a rail gun you should be able to tell this at a glance. If you have to be shot first then it's too late. 

Additionally, why could you not just take the codex definition approach with the current system? Based on what you say here you clearly aren't against information in the codex so why not have the barrel state what it's compatible with? This could easily be done with text or even with some form of icons. 

> The simple advantage of this system is that you see all the weapon options in front of you and what each of them does. Barrels and modules are there to slightly alter the weapon to your liking, if you want more range at the cost of X or more damage at the cost of Y.

The disadvantages though are plentiful:
1. No visual indication of what weapon they have until used
2. Way more codex clutter
3. Less freedom for the user when designing weapons
4. More confusion than the old system

> > As a whole, it just creates more confusion as now you have x number of weapon types buy only like 2 that use any of the addon systems.
> 
> And btw this remains the same in either system. Beam doesn't have barrels in current or my idea. So the issue is the same except in current you will have weapons that can't use any barrels, weapons that can only use some of the specialized barrels, and weapons that can use the universal barrels and only some of the specialized barrels. To figure out what weapons can use what barrels, you either have to read though a list or start picking ones at random and testing (the noobie approach) where in my idea you know by shot description if the weapon can use the universal barrel or not.

That is exactly my point.... In the current version there is one weapon type that can't use barrels. It's easy to understand. Your solution adds way more weapons types that may or may not use barrels. As such it compounds the issue of inconcistency. 

What's wrong with having to look at the details of the item you are using? If the user is unwilling to look at a short list then that sounds like their problem. There is a point when accessibility harms gameplay and this design seems to be getting to that point. 

> ![image](https://user-images.githubusercontent.com/36980446/120229507-21ef8300-c202-11eb-860e-abcec73079f7.png)
> Mine on top- Larger set of basic weapons. Only universal modules vs Fewer basic weapons, lots of barrels needed to get specialized weapons with who knows how odd of a chart of what can use what. Of course if it were the case it would have better names and such, but if you want variety this is what you will deal with

![image](https://user-images.githubusercontent.com/43644633/120232121-cbf2fd80-c24a-11eb-923f-42d6934624ba.png)
I fail to see how this is considered complicated or not user-friendly. If you have few core types and a variety of specialised add-ons then the logic is simple. 

You can add a list of maybe 2 or 3 core types and it's easy to use. 

Additionally, specialised weapons like rail guns or Gatling guns should have a trade-off. Otherwise more basic weapons like lasers and standard kinetics make no point. If you can have rail guns with range addons then what's the point in having a standard long-range kinetic weapon? Choosing a railgun should lock some other options. Designing your ship weapons should be a case of balancing pros and cons to find the ideal compromise for the purpose of your design. It should be simple to use but complicated to master. The current system needs more direction but this simplification takes away from the process of building in a building game.

**Comment by Kaiser-Indrasil** [comment]
My invaluable input:
Beam should be called Laser, because it's a line of photons moving at the speed of light. As in, hitscan.
Laser should be called Plasma or Hybrid, because it's a super-heated projectile (plasma) dealing a hybrid of heat and hull damage.
There, all the weapons are fixed.
Also, Cooling Barrel's OP AF, but that's a balancing issue.

**Comment by ZachZent** [comment]
Ok this is getting too long and is based off little information. Until Tsuna makes a decision on variety, there is nothing more to say
